### PR TITLE
296 reduce alphadia output

### PR DESCRIPTION
**Pull request by mschwoer**
- add options to save library files (@GeorgWa is `general` the right place?)
- reduce tqdm output

Plus some small unrelated changes:
- fix issue when ouput folder parents do not exist
- make error logging more robust
- add debug instructions
- allow to skip neptune upload for convenient local testing

### CODE REVIEW

File: alphadia/outputtransform.py [hunk 1]
```diff
@@ -365,7 +365,11 @@ def build(
...
         _ = self.build_lfq_tables(folder_list, psm_df=psm_df, save=True)
-        _ = self.build_library(base_spec_lib, psm_df=psm_df, save=True)
+        _ = self.build_library(
+            base_spec_lib,
+            psm_df=psm_df,
+            save=self.config["general"]["save_mbr_library"],
+        )
 
         if self.config["transfer_library"]["enabled"]:
             _ = self.build_transfer_library(folder_list, save=True)
```
Comment (mschwoer) [inline alphadia/outputtransform.py:375]: should we make saving those guys also optional?

File: alphadia/constants/default.yaml [hunk 2]
```diff
@@ -11,6 +11,9 @@ general:
   wsl: false
   mmap_detector_events: false
   use_gpu: true
+  # whether to save the (MBR) library to the output directory
+  save_library: True
```
Comment (GeorgWa) [inline alphadia/constants/default.yaml:15]: I think we could add that the first lib is the input library and the second is the output library.

File: alphadia/outputtransform.py (hunk 1 above)
Comment (GeorgWa) [reply alphadia/outputtransform.py:375]: Are they not already optional and only performed if transfer library enabled =True?

**Review by GeorgWa — APPROVED** [review]
I'm a bit opinionated about using self.config in the outputtransform  for consistency 😄
Otherwise it LGTM

File: alphadia/outputtransform.py (hunk 1 above)
Comment (mschwoer) [reply alphadia/outputtransform.py:375]: I was thinking  about the saving only ..

**Pull request merged**